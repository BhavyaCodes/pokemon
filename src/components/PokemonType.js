import React from "react";
import "../assets/css/badgeStyles.css";

import pokemonTypeToClassname from "../constants/pokemonTypeToClassname";

function PokemonType({ type }) {
  const getClassName = () => {
    return pokemonTypeToClassname[type] || "badge badge-pill badge-info mx-1";
  };

  if (!type) {
    return null;
  }
  return (
    <span data-testid={`pokemon-type-${type}`} className={getClassName()}>
      {type}
    </span>
  );
}

export default PokemonType;
