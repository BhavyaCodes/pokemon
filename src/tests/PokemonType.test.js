import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import PokemonType from "../components/PokemonType";

it("renders PokemonType component without crashing", () => {
  const mockType = "normal";

  const { asFragment, getByTestId } = render(<PokemonType type={mockType} />);
  expect(asFragment()).toMatchSnapshot();
  expect(getByTestId("pokemon-type-normal")).toHaveTextContent("normal");
});

it("renders null if type is not defined", () => {
  const { asFragment } = render(<PokemonType />);
  expect(asFragment()).toMatchSnapshot();
});
