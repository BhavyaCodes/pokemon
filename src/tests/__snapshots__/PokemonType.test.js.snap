// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PokemonType component without crashing 1`] = `
<DocumentFragment>
  <span
    class="badge badge-pill badge-normal mx-1"
    data-testid="pokemon-type-normal"
  >
    normal
  </span>
</DocumentFragment>
`;

exports[`renders null if type is not defined 1`] = `<DocumentFragment />`;
